File: movieApp/app/src/main/java/com/lbc/practice/movieapp/model/resource/remote/RemoteModule.java
package com.lbc.practice.movieapp.model.resource.remote;


import com.lbc.practice.movieapp.RetrofitManager;
import com.lbc.practice.movieapp.view.ActivityScope;

import dagger.Module;
import dagger.Provides;
import io.reactivex.disposables.CompositeDisposable;

@Module
public class RemoteModule {

    @RetroScope
    @Provides
    public RetrofitManager retrofitManager() {
        return new RetrofitManager();
    }

    @ActivityScope
    @Provides
    public CompositeDisposable disposeRetro() {
        return new CompositeDisposable();
    }
}

File: movieApp/app/src/main/java/com/lbc/practice/movieapp/data/resource/remote/RemoteModule.kt
package com.lbc.practice.movieapp.data.resource.remote;


import com.lbc.practice.movieapp.RetrofitManager;

import dagger.Module;
import dagger.Provides;
import io.reactivex.disposables.CompositeDisposable;

@Module
class RemoteModule {

    @Provides
    fun retrofitManager(): RetrofitManager {
        return RetrofitManager()
    }

    @Provides
    fun disposeRetro(): CompositeDisposable {
        return CompositeDisposable()
    }
}
